samza-core/src/main/scala/org/apache/samza/container/SamzaContainerMetrics.scala

```diff
@@ -49,6 +49,7 @@ class SamzaContainerMetrics(
   val executorWorkFactor = newGauge("executor-work-factor", 1.0)
   val physicalMemoryMb = newGauge("physical-memory-mb", 0.0F)
   val physicalMemoryUtilization = newGauge("physical-memory-utilization", 0.0F)
+  val totalProcessCpuUsage = newGauge("total-process-cpu-usage", 0.0)
```

Comment:
We are not rolling into the existing metrics is it? Why so? 
If we want to keep it separate, can we add documentation about the new metric to the website docs?

Reply:
Prefer to create a new metric(e.g. total-process-cpu-usage) to represent the cpu usage for the Samza container’s JVM process and all its child/descendant processes because of
- I view it as a non-backward compatible change if let the existing “process-cpu-usage” metric to include child/descendant processes especially for non-JVM processes.
  - Samza official doc clearly states that the `process-cpu-usage` returns the “recent cpu usage” for the Java Virtual Machine process: https://samza.apache.org/learn/documentation/1.6.0/operations/monitoring.html
- Having a new metric could be consistent with how Samza exposes physical-memory-mb . This memory related metric which includes child process is available for Samza container only, the AM doesn’t have it.
- Implementation/Maintainance complexity. If keep using the existing `process-cpu-usage` we will have to deprecate the JVMMetrics implementation and add a new hook into AM to expose the metric.

Good point on the doc, let me update the document.